src/metabase/driver/postgres.clj

```diff
@@ -770,3 +773,49 @@
   ;; This could be incorrect if Postgres has been compiled with a value for NAMEDATALEN other than the default (64), but
   ;; that seems unlikely and there's not an easy way to find out.
   63)
+
+(defn- format-copy
+  [_clause table]
+  [(str "COPY " (sql/format-entity table))])
+
+(sql/register-clause! ::copy format-copy :insert-into)
+
+(defn- format-from-stdin
+  [_clause delimiter]
+  [(str "FROM STDIN NULL " delimiter)])
+
+(sql/register-clause! ::from-stdin format-from-stdin :from)
+
+(defn- db->connection
+  [db-or-id]
+  (.unwrap (jdbc/get-connection (sql-jdbc.conn/db->pooled-connection-spec db-or-id))
+           PgConnection))
+
+(defn- sanitize-value
+  ;; Per https://www.postgresql.org/docs/current/sql-copy.html#id-1.9.3.55.9.2
+  [v]
+  (if (string? v)
+    (-> v
+        (str/replace "\\" "\\\\")
+        (str/replace "\n" "\\n")
+        (str/replace "\r" "\\r")
+        (str/replace "\t" "\\t"))
+    v))
+
+(defn- row->tsv
+  [row]
+  (->> row
+       (map sanitize-value)
+       (str/join "\t")))
+
+(defmethod driver/insert-into :postgres
+  [_driver db-id table-name column-names values]
+  (let [copy-manager (CopyManager. (db->connection db-id))
+        [sql & _]    (sql/format {::copy       (keyword table-name)
+                                  :columns     (map keyword column-names)
+                                  ::from-stdin "''"})
```

Comment:
Seems like it would be easier to write this without honeysql and use string manipulation instead

Reply:
I'm leaning on Honey SQL for sanitizing the table name, sanitizing (and joining together) the column names, and quoting, so I figured I may as well just stitch it all together in a Honey SQL clause :shrug:

Reply:
True, though quoting and joining column names is `(str/join "," (map metabase.driver.sql.util/quote-name column-names)), and sanitizing is only checking for semicolons.

```
(defn check-for-semicolons [s]
  (when (re-find #";" s)
    (throw (ex-info (tru "Semi-colons are not allowed. Found one in {0}" s) {}))))
```

Not a biggie but I think this would need fewer fns to jump through to understand what's going on